### DISCUSSION TITLE
GitHub not recognizing auto-generated MIT license

### DISCUSSION THREAD

**Discussion by safakgur** [community/community#22973] [New to GitHub]
<p>To get a license file, I followed the steps below:</p>
<ul>
<li>I clicked “Create new file” on the <a href="https://github.com/safakgur/guard" title="safakgur/guard: A high-performance, extensible argument validation library." rel="nofollow noopener">repo</a>’s main page.</li>
<li>I named the file “LICENSE.txt”, then clicked on the “Choose a license template” button.</li>
<li>I chose the MIT license, clicked the “Review and submit” button.</li>
</ul>
<p>I commited the exact output (not from the GitHub UI, I created the file locally and commited that) but GitHub doesn’t show the license in the repo header. It also doesn’t display the license summary when I open the <a href="https://github.com/safakgur/guard/blob/master/LICENSE.txt" rel="nofollow noopener">file itself</a>.</p>
<p>What can I do to make GitHub recognize my standard MIT license?</p>

**Comment by ghost** [comment]
<p>Hi there <a class="mention" href="/safakgur">@safakgur</a>,</p>
<p>That’s not cool, but I might be able to help you here. Does removing the  <strong>.txt</strong> extension help with this? :) </p>
<p>Let me know.</p>

**Comment by safakgur** [comment]
<p>Hi there, thank you for the answer.</p>
<p>I tried both removing the extension and adding a .md extension, neither worked.</p>

**Comment by ghost** [comment]
<p>Thanks for checking that, <a class="mention" href="/safakgur">@safakgur</a>.</p>
<p>Let’s see if removing the file locally &gt; pushing the change to GitHub &gt; creating the file using <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a> helps?</p>
<p>Just to make sure, when creating the file<a href="https://github.com/org/repo/new/branch?filename=LICENSE" rel="nofollow noopener">copy this URL</a> replacing the username/organisation, repository and branch with yours. Make sure to also press “Choose a license template” &gt; MIT</p>
<p>If that doesn’t work, just let me know and I’ll see what else I can suggest :) </p>

**Comment by safakgur** [comment]
<p>Thank you <a class="mention" href="/timrossback">@timrossback</a>, I just created a <a href="https://github.com/safakgur/guard/blob/license/LICENSE" rel="nofollow noopener">LICENSE</a> file exactly as you suggested and removed the old LICENSE.txt using the GitHub UI. Still nothing changes. One thing to note is that I’m not doing these experiments on master. I’m on a branch called license. I tried changing my default branch to license, however and that didn’t help.</p>

**Comment by ghost** [comment, marked as answer]
<p> Thanks for that <a class="mention" href="/safakgur">@safakgur</a>,</p>
<p>Would you be able to try creating a LICENSE file on the  <strong>master</strong> branch? If that doesn’t work, would you be able to try entering “Test Test Test” instead of “Şafak Gür” on <a href="https://github.com/safakgur/guard/blob/license/LICENSE#L3" rel="nofollow noopener">line 3 of the license</a>? :) </p>

**Comment by safakgur** [comment]
<p>Hi again, I didn’t want to touch the master branch of this repo so I’ve created a couple more to test. Turns out the license needs to be on the repo’s main branch but it’s ok for the main branch to be different than master. I thought I tested this case yesterday but apparently I did something wrong.</p>
<p>So it works with the license file I created and committed using the GitHub UI but not when I copy that license to a local file and commit+push from there. I wonder whether the files were the same, i.e. if there were some invisible differences like different line endings.</p>
<p>I’ll do a byte by byte comparison when I can, thank you for the suggestions.</p>

**Comment by ghost** [comment]
<p>Ah, that’s great news <a class="mention" href="/safakgur">@safakgur</a>! I’m so glad you got it sorted.</p>
<p>I’d also really suggest you also reach out to the folks over at GitHub support. They’ll be able to have a look backstage and see why this has happened (so it doesn’t happen again!) You can get in touch with them over here: <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a> :) </p>
<p>If you need a hand with anything, just let me know - I’m here for you.</p>

**Comment by mu578** [comment]
<p>Hello, I have the same issue here*;  did everything mentionned; move ; delete ; create. May we have a gitattribute for that? </p>
<p>* <a href="https://github.com/moe123/macadam" rel="nofollow noopener">https://github.com/moe123/macadam</a></p>
